### ISSUE TITLE
DAC18 object detection dataset

### ISSUE THREAD

**Issue by tangrui2018** [fpgasystems/spooNN#5]
Hi,
Now I want to train this to our own data set. But I can't recurrence your project. Could you please give us directions to train this to our own data set? I know DAC18 object detection dataset is not publicly available yet. Or could you please give me your DAC18 object detection dataset?

**Comment by kaankara** [comment]
I am afraid I can't provide the DAC18 data set. How to train with your own data set depends on what type of data that is. Are you also doing object detection? Do you have images with bounding-box labels? If so, feeding them to the existing training script should be relatively straightforward. Just check how the data is read using Tensorpack, how it is resized etc. If you are dealing with a classification problem, then you need to change the loss function etc. in the training script, which is a bit more work.

**Comment by Gchinanty** [comment]
Hello, first of all, thank you for your work, and my colleagues and I all think it is a great repo.
I have a question, If my dataset is VOC format and I want to reproduce your repo, would you please tell me how I need to do?

**Comment by Gchinanty** [comment]
@kaankara 
Hello, first of all, thank you for your work, and my colleagues and I all think it is a great repo.
I have a question, If my dataset is VOC format and I want to reproduce your repo, would you please tell me how I need to do?

**Comment by tangrui2018** [comment]
Thank you very much. I will take many detours without you. My dataset is VOC format and I want to do object detection to it. Could you please tell me what I need to do?

**Comment by kaankara** [comment]
This will require some changes to the training script. In halfsqueezenet/training/halfsqueezenet_objdetect.py, please take a look at the DAC_Dataset class. This class is responsible for reading the images and bounding box labels from .xml files. You basically should write your own class that can read data in the format of VOC data set. After you do this, if you generate the same type of labels as DAC_Dataset class, you can use the model (this is the neural network) without any changes. Keep in mind that the existing model is responsible for detecting only one object on an image, whereas in VOC there might be multiple objects to be detected. If you actually want to detect multiple objects, you need to also change the model.
